### PR TITLE
Fix intel-mkl package

### LINKED ISSUE #14855: installation issue: intel-mkl
It seems like `intel-mkl` needs the command line program `cpio` in order to build.  In a Docker image, e.g. `spack/ubuntu-bionic`, where `cpio` is not installed, I'm getting the following error:

```console
[ubuntu] ~: spack install -d intel-mkl
...
There are one or more critical unresolved issues which prevent setup from
continuing.  Fix them and run the setup program again.
--------------------------------------------------------------------------------
Missing critical prerequisite
-- 'cpio' tool could not be found
The setup program requires 'cpio' tool to complete installation. Please make
sure this tool is installed and its location is present in 'PATH' environment
variable.
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
==> [2020-02-04-23:31:45.828090] WRITE LOCK: /super/absurdly/ridiculously/obnoxiously/long/path/for/relocation/.spack-db/prefix_lock[6645195540281252399:1] [Released]
==> [2020-02-04-23:31:45.828348] WRITE LOCK: /tmp/root/spack-stage/.lock[493556182836919143:1] [Released]
==> [2020-02-04-23:31:45.840027] Error: ProcessError: Command exited with status 4:
    './install.sh' '--silent' 'silent.cfg'

1 warning found in build log:
  >> 4    WARNING: Destination directory already exists.
     5    
     6    There are one or more critical unresolved issues which prevent setup 
          from
     7    continuing.  Fix them and run the setup program again.
     8    ---------------------------------------------------------------------
          -----------
     9    Missing critical prerequisite
     10   -- 'cpio' tool could not be found

See build log for details:
  /tmp/root/spack-stage/spack-stage-intel-mkl-2020.0.166-xbypirns6k2f5cow3wzxqa3aplktxpiv/spack-build-out.txt
...
```

### PR DESCRIPTION
**Pull request by scottwittenburg**
This PR adds a `cpio` package and makes it available during the build of `intel-mkl`.

Fixes #14855

### CODE REVIEW

**Comment by scottwittenburg** [comment]
Fixes #14855

File: var/spack/repos/builtin/packages/intel-mkl/package.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ class IntelMkl(IntelPackage):
     version('11.3.2.181', sha256='bac04a07a1fe2ae4996a67d1439ee90c54f31305e8663d1ccfce043bed84fc27',
             url='http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/8711/l_mkl_11.3.2.181.tgz')
 
+    depends_on('cpio')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-mkl/package.py]: type='build'

File: var/spack/repos/builtin/packages/intel-mkl/package.py [hunk 2]
```diff
@@ -72,3 +74,6 @@ class IntelMkl(IntelPackage):
     if sys.platform == 'darwin':
         # there is no libmkl_gnu_thread on macOS
         conflicts('threads=openmp', when='%gcc')
+
+    def setup_build_environment(self, env):
+        env.prepend_path('PATH', self.spec['cpio'].prefix.bin)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-mkl/package.py]: This shouldn’t be necessary, build deps are automatically added to the PATH

File: var/spack/repos/builtin/packages/intel-mkl/package.py (hunk 1 above)
Comment (scottwittenburg) [reply var/spack/repos/builtin/packages/intel-mkl/package.py]: Done, thanks for the review @adamjstewart.

File: var/spack/repos/builtin/packages/intel-mkl/package.py (hunk 2 above)
Comment (scottwittenburg) [reply var/spack/repos/builtin/packages/intel-mkl/package.py]: Fixed, thanks.

**Comment by scottwittenburg** [comment]
Thanks @adamjstewart!

**Pull request merged**